playbooks/tasks/portal-role-task-mongo-setup-docker-service.yml

```diff
@@ -0,0 +1,126 @@
+  ansible.builtin.stat:
+    path: "{{ mongo_mgkey_file }}"
+  register: mgkey_stat_result
+
+- name: Ensure mongo/mgkey file content is up-to-date
+  ansible.builtin.copy:
+    dest: "{{ mongo_mgkey_file }}"
+    content: "{{ webportal_common_config.mongo_db_mgkey }}"
+  become: True
+  # Perform only if mongo/mgkey is a regular file (not a dictionary)
+  when: mgkey_stat_result.stat.isreg | default(False)
+  register: mgkey_update_result
+
+- name: Include checking mongo service is running correctly
+  include_tasks: tasks/portal-role-task-mongo-check-docker-service.yml
+  when: mgkey_stat_result.stat.isreg | default(False) and not mgkey_update_result.changed
+
+- name: Log mongo service status if not ok
+  debug:
+    msg: |
+      Mongo service does NOT run correctly.
+      We will reset the mongo service.
+  when: not (mongo_service_ok | default(False))
+
+# Reset mongo service if it is not running correctly
+- block:
+    - name: Stop and remove mongo container
+      community.docker.docker_container:
+        name: mongo
+        state: absent
+        container_default_behavior: no_defaults
+
+    - name: Remove mongo/db directory
+      ansible.builtin.file:
+        path: "{{ mongo_db_dir }}"
+        state: absent
+      become: True
+    
+    - name: Create empty mongo/db directory
+      ansible.builtin.file:
+        path: "{{ mongo_db_dir }}"
+        state: directory
+        owner: "999"
+        group: "999"
+      become: True
+
+    - name: Remove mongo/mgkey file/directory
+      ansible.builtin.file:
+        path: "{{ mongo_mgkey_file }}"
+        state: absent
+      become: True
+
+    - name: Create mongo/mgkey file
+      ansible.builtin.copy:
+        dest: "{{ mongo_mgkey_file }}"
+        content: "{{ webportal_common_config.mongo_db_mgkey }}"
+        owner: "999"
+        group: "999"
+        mode: "0400"
+      become: True
+
+    # Get MongoDB image version from docker-compose.mongodb.yml file
+
+    - name: Read docker-compose.mongodb.yml
+      slurp:
+        src: "{{ webportal_dir }}/docker-compose.mongodb.yml"
+      register: docker_compose_mongodb_slurp
+
+    - name: Parse docker-compose.mongodb.yml
+      set_fact:
+        docker_compose_mongodb_vars: "{{ docker_compose_mongodb_slurp['content'] | b64decode | from_yaml }}"
+
+    - name: Get mongo image version
+      set_fact:
+        mongodb_image_version: "{{ docker_compose_mongodb_vars.services.mongo.image }}"
+
+    # Initialize MongoDB
+
+    - name: Initialize mongo database with admin credentials
+      community.docker.docker_container:
+        image: "{{ mongodb_image_version }}"
+        name: mongo-setup
+        command: "-keyFile=/data/mgkey --replSet={{ webportal_common_config.skynet_db_replicaset }}"
+        env:
+          MONGO_INITDB_ROOT_USERNAME: "{{ mongodb_config.skynet_db_user }}"
+          MONGO_INITDB_ROOT_PASSWORD: "{{ mongodb_config.skynet_db_pass }}"
+        published_ports: "27017:27017"
```

Comment:
could re-use `mongodb_config.skynet_db_port` right? `"{{ mongodb_config.skynet_db_port }}:{{ mongodb_config.skynet_db_port }}"`

Reply:
Extracted to `default_mongo_port`. It's a Skynet webportal constant, not a configurable value.